api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java

```diff
@@ -84,7 +90,7 @@ private static ItemStack itemNotNull(ItemStack stack) {
     /** An {@link Consumer} that is called when the anvil GUI is close */
     private final Consumer<StateSnapshot> closeListener;
     /** An {@link BiFunction} that is called when a slot is clicked */
-    private final BiFunction<Integer, StateSnapshot, List<ResponseAction>> clickHandler;
+    private final BiFunction<Integer, StateSnapshot, CompletableFuture<List<ResponseAction>>> clickHandler;
```

Comment:
This type feels like its getting messy. I think we should just add a new subclass:
```java
public interface ClickHandler extends BiFunction<Integer, StateSnapshot, CompletableFuture<List<ResponseAction>>> {
}
```
Then replace all of our verbose `BiFunctions` with `ClickHandler`.